backend/drm/drm.c

```diff
@@ -197,36 +197,28 @@ static bool init_planes(struct wlr_drm_backend *drm) {
 			goto error;
 		}
 
-		/*
-		 * This is a very naive implementation of the plane matching
-		 * logic. Primary and cursor planes should only work on a
-		 * single CRTC, and this should be perfectly adequate, but
-		 * overlay planes can potentially work with multiple CRTCs,
-		 * meaning this could return inefficient/skewed results.
-		 *
-		 * However, we don't really care about overlay planes, as we
-		 * don't support them yet. We only bother to keep basic
-		 * tracking of them for DRM lease clients.
-		 *
-		 * possible_crtcs is a bitmask of crtcs, where each bit is an
-		 * index into drmModeRes.crtcs. So if bit 0 is set (ffs starts
-		 * counting from 1), crtc 0 is possible.
-		 */
-		int crtc_bit = ffs(plane->possible_crtcs) - 1;
-
-		// This would be a kernel bug
-		assert(crtc_bit >= 0 && (size_t)crtc_bit < drm->num_crtcs);
+		// We don't really care about overlay planes, as we don't support them
+		// yet.
+		if (type == DRM_PLANE_TYPE_OVERLAY) {
+			continue;
```

Comment:
should drmModeFreePlane be called here? It is called in master.

Reply:
Good catch, fixed!